### PR TITLE
Add logger to show info

### PR DESCRIPTION
**Pull request by bloodbare**

### CODE REVIEW

File: nuclia/sdk/upload.py [hunk 1]
```diff
@@ -272,8 +272,8 @@ def remote(
                 for _ in tqdm(range((size // CHUNK_SIZE) + 1)):
                     chunk = r.raw.read(CHUNK_SIZE)
                     offset = ndb.patch_tus_upload(upload_url, chunk, offset)
-            except Exception as e:
-                print(e)
+            except Exception:
+                logger.exception("Error on uploading")
```
Comment (lferran) [inline nuclia/sdk/upload.py:276]: A better error message would be:
```python
                logger.exception("Error uploading")
```
or
```python
                logger.exception("Upload error")
```

Comment (lferran) [reply nuclia/sdk/upload.py:276]: same for all other occurrences in this same file

File: nuclia/sdk/search.py [hunk 2]
```diff
@@ -468,6 +484,9 @@ async def ask_json(
                 ask_response_item = AskResponseItem.model_validate_json(line).item
             except Exception as e:
                 warnings.warn(f"Failed to parse AskResponseItem: {e}. item: {line}")
+                import pdb
+
+                pdb.set_trace()
```
Comment (lferran) [inline nuclia/sdk/search.py:489]: breakpoint!

File: nuclia/sdk/search.py [hunk 3]
```diff
@@ -192,23 +192,37 @@ def ask_json(
...
+                raise
         elif isinstance(query, AskRequest):
             req = query
+        elif query is None:
+            req = AskRequest(
+                query="",
+                answer_json_schema=schema_json,
+            )
+            if filters is not None:
+                req.filters = filters
```
Comment (lferran) [inline nuclia/sdk/search.py:223]: Can you add tests for this?

Comment (lferran) [reply nuclia/sdk/search.py:223]: maybe unit tests is enough

File: nuclia/sdk/search.py [hunk 4]
```diff
@@ -192,23 +192,37 @@ def ask_json(
         See https://docs.nuclia.dev/docs/api#tag/Search/operation/Ask_Knowledge_Box_kb__kbid__ask_post
         """
         ndb: NucliaDBClient = kwargs["ndb"]
+        if isinstance(schema, str):
+            if os.path.exists(schema):
```
Comment (lferran) [inline nuclia/sdk/search.py:196]: what if the file does not exist? should we raise an error?

File: nuclia/sdk/search.py [hunk 5]
```diff
@@ -192,23 +192,37 @@ def ask_json(
...
+                with open(schema, "r") as json_file_handler:
+                    schema_json = json.load(json_file_handler)
```
Comment (lferran) [inline nuclia/sdk/search.py:198]: should we try/except here and return a proper error log if the contents of the file could not be loaded?

File: nuclia/sdk/search.py [hunk 6]
```diff
@@ -172,17 +173,16 @@ def ask(
...
+                result.timings = ask_response.metadata.timings.model_dump()
             if ask_response.metadata.tokens is not None:
-                result.tokens = ask_response.metadata.tokens.dict()
+                result.tokens = ask_response.metadata.tokens.model_dump()
         return result
 
     @kb
     def ask_json(
         self,
-        *,
```
Comment (lferran) [inline nuclia/sdk/search.py:183]: I don't really mind this change but all other methods have the `*` that forces users to use named params.

File: nuclia/sdk/kb.py [hunk 7]
```diff
@@ -198,23 +165,13 @@ def __init__(self):
         self.imports = AsyncNucliaImports()
 
     @kb
-    async def list(
-        self, *, interactive: bool = True, **kwargs
-    ) -> Optional[ResourceList]:
+    async def list(self, *, interactive: bool = True, **kwargs) -> ResourceList:
```
Comment (lferran) [inline nuclia/sdk/kb.py:168]: should the `interactive` param be removed like above in the sync version?

File: nuclia/sdk/resource.py [hunk 8]
```diff
@@ -47,6 +51,61 @@ def create(*args, **kwargs) -> str:
...
+        elif isinstance(query, dict):
+            try:
+                req = AskRequest.model_validate(query)
+            except ValidationError:
+                raise ValueError("Invalid AskRequest object.")
+
+        elif isinstance(query, AskRequest):
+            req = query
+        else:
+            raise ValueError("Invalid query type. Must be str, dict or AskRequest.")
```
Comment (lferran) [inline nuclia/sdk/resource.py:90]: some unit tests would be nice on this param parsing. and Maybe add some more error handling for the loading from file path

**Review by ebrehault — APPROVED** [review]

**Pull request merged**